savoirfairelinux/sous-chef
# Fixed #610: support walking/driving/cycling in leaflet routing js

## Fixes #610  by lingxiaoyang

### Changes proposed in this pull request:

* Add three vehicle buttons on leaflet routing interface. Uses Semantic UI basic button group.

![image](https://cloud.githubusercontent.com/assets/8630726/21818870/7415767c-d737-11e6-943e-927ab99f8523.png)


* Refactor the related JS code to be conformed to Leaflet Routing API documentation.

* Add required model field `Route.vehicle` (default to "cycling"). When the user changes vehicle in leaflet JS, this field is updated in the backend.

* Change the vehicle icon in route overview corresponding to `Route.vehicle`: (driving/cycling/walking)
![image](https://cloud.githubusercontent.com/assets/8630726/21821975/78fc146e-d743-11e6-9317-92d02554ca08.png)

* Unit tests for SaveRouteView and SaveRouteVehicleView.

### Status


### How to verify this change

Set up kitchen count with a few geolocalized clients. At the last step "Organize Routes", it should be functional. The route should change when changing the vehicle. Then go back to "Route overview", the vehicle icon should be changed. Then go to django admin, find the route, the "vehicle" field should be changed.

### Deployment notes and migration

 - [x] DB Migration is needed.

### New translatable strings


### Additional notes

none
